src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public Integer executeOperation(FruitDao fruitDao, FruitTransaction fruitTransaction) {
+        return fruitDao.putFruits(fruitTransaction.getFruit(), -fruitTransaction.getQuantity());
+    }
```

Comment:
The implementation of `executeOperation` in `PurchaseOperation` does not check if the result balance is positive before saving it in the storage as required by the checklist. You should add a check to ensure that the balance cannot be negative after a purchase operation. If the balance would be negative, throw a `RuntimeException`.